curation-migration-datasync/eventConsumer.ts

```diff
@@ -55,3 +55,43 @@ export async function removeScheduledItem(
   // Remove association from DynamoDB
   await curatedItemModel.deleteByCuratedRecId(curatedRecord.curatedRecId);
 }
+
+/**
+ * fetches all curatedRecId related to the approvedItem's externalId.
+ * and updates them.
+ * @param eventBody
+ */
+export async function updatedApprovedItem(
+  eventBody: ApprovedItemPayload,
+  db: Knex
+) {
+  // dynamoDb will have a record of curatedRecId mapped to approvedItem
+  // only if it was previously scheduled.
+  const curatedItemModel = new CuratedItemRecordModel();
+  const curatedItems = await curatedItemModel.getByApprovedItemExternalId(
+    eventBody.approvedItemExternalId
+  );
+
+  //if dynamo returns 0 curatedItem, then the approvedItem was not scheduled before,
+  //so we can safely ignore this event.
+  if (curatedItems.length == 0) {
+    return;
+  }
+
+  //if there are approvedItem in the dynamoDb, they must be scheduled before.
+  //all scheduled item has a record in our legacy database.
+  // so we update them as per the new eventBody.
+  for (const curatedItem of curatedItems) {
+    try {
+      const dbService = new DataService(db);
+      await dbService.updateApprovedItem(eventBody, curatedItem.curatedRecId);
+      //update lastUpdatedAt alone in dynamoDb.
+      curatedItem.lastUpdatedAt = Math.round(new Date().getTime() / 1000);
+      await curatedItemModel.insert(curatedItem);
+    } catch (e) {
+      throw new Error(
```

Comment:
Should we stop the whole loop if an ID isn't in the db? I wonder if it's better to just skip any records that don't exist, since it has no implication for data sync if they aren't in the db.

Reply:
we do ignore if the ID doesn't exist in the dynamo at all,

 this error get's thrown when the ID is in dynamoDb but not in the database, i feel if the Id somehow made it to dynamo but if it doesn't exist in the db, its concerning how it got to dynamo (or why it was not deleted with delete-evens) and we should probably look into it. 

(or) are you saying we need to just throw an error when this `item` is empty, and not proceed with following transactions? https://github.com/Pocket/curation-tools-data-sync/blob/7f9a6d2bacc733d20512efd7914c67dade9cfa6f/curation-migration-datasync/database/dataService.ts#L148

Reply:
logging error if curatedRecId is not found